features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapListenerConfig.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2021 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2021 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.trapd;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import org.opennms.netmgt.config.TrapdConfig;
+import org.opennms.netmgt.snmp.SnmpV3User;
+
+import com.google.common.base.MoreObjects;
+
+@XmlRootElement(name = "trap-listener-config")
+@XmlAccessorType(XmlAccessType.NONE)
+public class TrapListenerConfig {
```

Comment:
While working on adding Karaf command line, I thought it would be better to move this into   `org.opennms.core.snmp.api`  like `SnmpV3User`. That would make Karaf command module to be independent of specific modules.

Reply:
> While working on adding Karaf command line, I thought it would be better to move this into `org.opennms.core.snmp.api` like `SnmpV3User`. That would make Karaf command module to be independent of specific modules.

I would expect the karaf command to not depend on the specific DTOs but just print out JSON or something similar.

Reply:
When we need to subscribe, we need to pass specific class
`subscribe(String key, Class<T> clazz, Consumer<T> consumer)`

That's where it is required